tests/Unit/Security/Lti/LoginHintValidatorTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Tests\Unit\Security\Lti;
+
+use OAT\SimpleRoster\DataTransferObject\LoginHintDto;
+use OAT\SimpleRoster\Entity\User;
+use OAT\SimpleRoster\Exception\LineItemNotFoundException;
+use OAT\SimpleRoster\Lti\Exception\InvalidGroupException;
+use OAT\SimpleRoster\Lti\Extractor\LoginHintExtractor;
+use OAT\SimpleRoster\Repository\UserRepository;
+use OAT\SimpleRoster\Security\Lti\LoginHintValidator;
+use OAT\SimpleRoster\Tests\Traits\DatabaseTestingTrait;
+use PHPUnit\Framework\MockObject\MockObject;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class LoginHintValidatorTest extends KernelTestCase
+{
+    use DatabaseTestingTrait;
+
+    /** @var LoginHintValidator */
+    private $subject;
+
+    /** @var LoginHintExtractor|MockObject */
+    private $loginHintExtractor;
+
+    /** @var UserRepository|MockObject */
+    private $userRepository;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        self::bootKernel();
+
+        $this->loginHintExtractor = $this->createMock(LoginHintExtractor::class);
+        $this->userRepository = $this->createMock(UserRepository::class);
+
+        $this->subject = new LoginHintValidator($this->loginHintExtractor, $this->userRepository);
+
+        $this->setUpDatabase();
+        $this->loadFixtureByFilename('userWithReadyAssignment.yml');
+    }
+
+    public function testLoginHintIsValid(): void
+    {
+        $user = $this->getRepository(User::class)->find(1);
+
+        $loginHint = 'user1::group_1::lineItemSlug';
+        $loginHintDto = new LoginHintDto('user1', 'group_1', 'lineItemSlug');
+
+        $this->userRepository
+            ->expects($this->once())
+            ->method('findByUsernameWithAssignments')
+            ->with('user1')
+            ->willReturn($user);
+
+        $this->loginHintExtractor
+            ->expects($this->once())
+            ->method('extract')
+            ->with($loginHint)
+            ->willReturn($loginHintDto);
+
+        $result = $this->subject->validate($loginHint);
+
+        $this->assertSame($result, $user);
+    }
```

Comment:
Again, it's a bit inconsistent (though may be the only choice) to have `testLoginHintIsValid` which is calling `validate` method and not asserting returned value is boolean true, but objects are the same

Reply:
Removed the validator, checking login hint consistency directly on OidcUserAuthenticator

Reply:
We don't have the validator anymore